### PR TITLE
[IMP] l10n_pl: fill accounting date with delivery date

### PR DESCRIPTION
**Pull request by MuhamedAbdallaMuhamed**
It seems that in Eastern European country, they need to use the delivery date as accounting date in majority of cases

solution: So the idea is to use the same feature that we have now but  use the delivery date as reference
in place of invoice date in the sales (always) and in the purchase if delivery date is earlier than invoice date

Also, the currency conversion rate is dependent on accounting date
but It seems that in Eastern European country needs to use the delivery date as currency rate date in the first place

Solution: the currency rate now is dependent on delivery date in first place

Task [link](https://www.odoo.com/odoo/project/967/tasks/3953743)
task-3953743

### CODE REVIEW

**Comment by muab-odoo** [comment]
@lordkrandel  Can you please take a look on here?
It has been there for long time :')
Thank!

**Review by gawa-odoo — COMMENTED** [review]
Hello!
Sorry for the delay, I hesitated on the fact that we did the change in 17.0 (wasn't certain with the stable policy, it's a change of behaviour), then saw that the task wasn't in my pipe, so I thought someone else took care of it.

Mainly the traceback that needs to be fixed.
The rest is if you want, not mandatory

File: addons/l10n_pl/models/account_move.py [hunk 1]
```diff
@@ -16,3 +16,23 @@ class AccountMove(models.Model):
...
+
+    @api.depends('delivery_date')
+    def _compute_date(self):
+        super()._compute_date()
+        for move in self:
+            if move.country_code == 'PL' and move.delivery_date:
+                if move.invoice_filter_type_domain == 'sale':
+                    move.date = move.delivery_date
+                elif move.invoice_filter_type_domain == 'purchase':
+                    move.date = min(move.delivery_date, move.invoice_date)
```
[the remark this answers is not in the document]
Comment (muab-odoo) [reply addons/l10n_pl/models/account_move.py]: Fixed!
Thanks for catching it.

File: addons/l10n_pl/models/account_move.py [hunk 2]
```diff
@@ -16,3 +16,27 @@ class AccountMove(models.Model):
...
+    def _compute_date(self):
+        # EXTENDS 'account'
+        super()._compute_date()
+        for move in self:
+            if move.country_code == 'PL' and move.delivery_date:
+                if move.invoice_filter_type_domain == 'sale':
+                    move.date = move.delivery_date
+                elif move.invoice_filter_type_domain == 'purchase':
+                    move.date = min(move.delivery_date, move.invoice_date or fields.Date.context_today(self))
+                # purchase/sale shouldn't have a delivery_date
```
Comment (gawa-odoo) [inline addons/l10n_pl/models/account_move.py:38]:
```suggestion
                # non purchase/sale shouldn't have a delivery_date
```
After that, LGTM!

Comment (muab-odoo) [reply addons/l10n_pl/models/account_move.py]: Done

**Comment by gawa-odoo** [comment]
Thanks 
@robodoo r+

**Comment by d-fence** [comment]
robodoo r-
seems to [break the staging](https://runbot.odoo.com/runbot/build/66672921)

**Review by lordkrandel — CHANGES_REQUESTED** [review]
Notice that we're having side-effects.

In the `l10n_standalone` build I see this:
immagine

which looks like a non-polish company's having its dates changed... ?

**Comment by muab-odoo** [comment]
> Notice that we're having side-effects.
> 
> In the `l10n_standalone` build I see this: immagine
> 
> which looks like a non-polish company's having its dates changed... ?

Yes, I know that and I'm trying for a long time to just import the l10n db on my side in order to see what is happening.
but I'm facing series of issue that is going to infiniity with me.

**Comment by gawa-odoo** [comment]
Mhhh I will need to check further, but it seems the precompute is destroying the compute done, as the l10n build is still fuckd :/
I'm not sure what exactly the issue is really

**Comment by lordkrandel** [comment]
The Chilean l10n_cl_edi test doesn't fail on my machine.
Investigating more, I rebased and will try again.
In case maybe the deleted branch has something to do with it, I can create a new duplicate one and see if it's any different.

**Pull request closed without merging**